### PR TITLE
Default rather than option

### PR DESCRIPTION
**Pull request by gilescope**
Implementes simplification suggested in #9177

I'm in two minds as to whether I like this change or not, so throwing it open to the reviewers.

### CODE REVIEW

File: client/finality-grandpa/src/environment.rs [hunk 1]
```diff
@@ -1173,7 +1173,7 @@ where
 	debug!(target: "afg", "Finding best chain containing block {:?} with number limit {:?}", block, limit);
```
Comment (gilescope) [inline client/finality-grandpa/src/environment.rs:1174]: just above it's looking for the block so we know it exists.

File: client/service/test/src/client/mod.rs [hunk 2]
```diff
@@ -450,28 +450,6 @@ fn best_containing_with_genesis_block() {
...
-		.build()
-		.unwrap()
-		.block;
-
-	assert_eq!(
-		None,
-		block_on(longest_chain_select.finality_target(uninserted_block.hash().clone(), None))
 			.unwrap(),
 	);
 }
```
Comment (gilescope) [inline client/service/test/src/client/mod.rs:455]: This test didn't really make sense any more.

**Comment by andresilva** [comment]
There is already #9192 open for this, since it's coming from a community member (and it also was opened first) I'd prefer merging that one instead. I haven't reviewed it yet since I was out, will do.

**Comment by gilescope** [comment]
Sorry missed that. Looks like we agree on the changes - their PR LGTM.

**Pull request closed without merging**